packages/form-integrations/test/form-validation-integrations.test.js

```diff
@@ -117,5 +118,63 @@ describe('Form Validation Integrations', () => {
       await el.feedbackComplete;
       expect(spy.called).to.be.false;
     });
+
+    it('correctly renders localized fieldName (label), even on locale changes', async () => {
+      class DefaultLabelInput extends LionInput {
+        constructor() {
+          super();
+          this.setDefaultLabel();
+        }
+
+        async setDefaultLabel() {
+          async function update() {
+            await localize.loadingComplete;
+            this.label = localize.msg('test-default-label:label');
+          }
+          const boundUpdate = update.bind(this);
+          localize.loadNamespace({
+            'test-default-label': /** @param {string} locale */ async locale => {
+              switch (locale) {
+                case 'nl-NL':
+                  return { label: 'Tekst' };
+                default:
+                  return { label: 'Text' };
+              }
+            },
+          });
+          boundUpdate();
+          localize.addEventListener('localeChanged', () => {
+            boundUpdate();
+          });
+        }
+      }
+      const elTagString = defineCE(DefaultLabelInput);
+      const elTag = unsafeStatic(elTagString);
+      const el = /** @type {LionInput} */ (
+        await fixture(html`
+        <${elTag}
+          .validators=${[new Required()]}
+        >${lightDom}</${elTag}>
+      `)
+      );
+      el.touched = true;
+      el.dirty = true;
+      await el.updateComplete;
+      await el.feedbackComplete;
+      const { _feedbackNode } = getFormControlMembers(el);
+      expect(_feedbackNode.feedbackData?.[0].message).to.equal('Please enter a(n) Text.');
+
+      localize.locale = 'nl-NL';
+      await el.updateComplete;
+      await el.feedbackComplete;
+      expect(el.label).to.equal('Tekst');
+      expect(_feedbackNode.feedbackData?.[0].message).to.equal('Vul een Tekst in.');
```

Comment:
This was failing because "Tekst" was still "Text" from the old locale. Now that it awaits localize.loadingComplete, it will get the right localized label in the feedback message